cluster/vagrant/util.sh

```diff
@@ -295,7 +295,17 @@ function kube-up {
    vagrant ssh master -- sudo cat /srv/kubernetes/kubecfg.key >"${KUBE_KEY}" 2>/dev/null
    vagrant ssh master -- sudo cat /srv/kubernetes/ca.crt >"${CA_CERT}" 2>/dev/null
 
+   # Update the user's kubeconfig to include credentials for this apiserver.
    create-kubeconfig
+
+   if [[ "${FEDERATION:-}" == "true" ]]; then
```

Comment:
nit: Given that this piece of code is duplicated verbatim for 4 (and in general 'n') cloud providers, would it not be better to place it in a common, shared library, and only call that from here?